src/libraries/System.Memory/src/System/Buffers/ReadOnlySequence.cs

```diff
@@ -528,6 +528,56 @@ public SequencePosition GetPosition(long offset)
             return Seek(offset);
         }
 
+        /// <summary>
+        /// Translate opaque <see cref="System.SequencePosition"/> value to numerical position offset
```

Comment:
The way this is written, it gives the impression that any arbitrary `SequencePosition` can be passed in and is valid. Also, using `translate` seems strange to me.

We should try to be consistent with existing docs:
https://docs.microsoft.com/en-us/dotnet/api/system.buffers.readonlysequence-1.getposition?view=netcore-3.1

How about:
```suggestion
        /// Returns the offset of a <paramref name="position" /> within this sequence from the start.
```

We should also add a remarks section stating that callers should only pass in a `SequencePosition` that belong to this `ReadOnlySequence`. Otherwise, the behavior is undefined.

Reply:
The other thing that might be worth noting in the remarks (and testing for) is that the behavior is "exclusive", not "inclusive".

Let's say we have one of these sequences:
[50] -> [0]
[50] -> [50]

What does calling `GetOffset(new SequencePosition(second segment, 0))` return? 50, correct? And then if I slice the ReadOnlySequence from 0 to that offset (i.e. 50), is the returned `ReadOnlySequence`, a single segment?

Something like:
```C#
ReadOnlySequence<byte> slice = ros.GetOffset(new SequencePosition(segment2, 0));
Assert.True(slice.IsSingleSegment);
```

Reply:
@ahsonkhan 

```cs
slice = ros.Slice(new SequencePosition(segment1, 50)); // This would be the exclusive case.
Assert.True(slice.IsSingleSegment);
```

this is not a single segment...the returned sequence is a multi segment starting from 50th item.